Incorrect paths in .pre-commit-config.yaml skip out on tests.

A regression introduced with PR https://github.com/jenstroeger/python-package-template/pull/27: the git hooks didn’t run their tests over the package source anymore because the source’s moved into the `src/` folder.

For example,

https://github.com/jenstroeger/python-package-template/blob/eaa6a604e739e1667ae390dc69cafa576dea597e/.pre-commit-config.yaml#L40-L42

uses a [regex for the `files` entry](https://pre-commit.com/#regular-expressions) and matches that regex against path names. That regex wouldn’t match anymore for the new `src/package/…` paths and therefore files wouldn’t be picked for the hook anymore. Not sure that’s actually fixable on the [pre-commit](https://github.com/pre-commit/pre-commit) side because the matched list of files is not empty — it’s just not complete either 🤔

I also noticed that [mypy](https://github.com/python/mypy) would generate an error
```
tests/test_something.py:5:1: error: Skipping analyzing "package.something": found module but no type hints or library stubs
tests/test_something.py:5:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 2 source files)
```
which can be fixed by adding [mypy_path](https://mypy.readthedocs.io/en/stable/config_file.html#confval-mypy_path) to the configuration.

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -38,7 +38,7 @@ repos:
     rev: v2.29.0
     hooks:
       - id: pyupgrade
-        files: ^package/|^tests/|setup.py
+        files: ^src/package/|^tests/|setup.py
         args: [--py39-plus]
 
   # Similar to pylint, with a few more/different checks.
@@ -46,7 +46,7 @@ repos:
     rev: 4.0.1
     hooks:
     - id: flake8
-      files: ^package/|^tests/|setup.py
+      files: ^src/package/|^tests/|setup.py
       additional_dependencies: [flake8-builtins==1.5.3,flake8-docstrings==1.6.0,flake8-rst-docstrings==0.2.3,pep8-naming==0.12.1]
 
   # Run Pylint from the local repo to make sure venv packages
@@ -57,7 +57,7 @@ repos:
         name: pylint
         entry: pylint
         language: python
-        files: ^package/|^tests/|setup.py
+        files: ^src/package/|^tests/|setup.py
         args: []
 
   # Type-check all Python code.
@@ -65,7 +65,7 @@ repos:
     rev: v0.910-1
     hooks:
       - id: mypy
-        files: ^package/|^tests/|setup.py
+        files: ^src/package/|^tests/|setup.py
         additional_dependencies: [types-setuptools==57.4.2,hypothesis==6.24.2]
         args: [--config-file, mypy.ini]
 

--- a/mypy.ini
+++ b/mypy.ini
@@ -5,6 +5,7 @@
 # get consistent results on any platform.
 python_version=3.9
 platform=linux
+mypy_path = src
 
 # Useful settings.
 show_column_numbers=True

